mitodl/open-discussions
# Only index lists with items, include item image_src values

#### Pre-Flight checklist

#### What are the relevant tickets?
Fixes #2437 
Fixes #2447

#### What's this PR do?
- Indexes lists only if they have items
- Deindexes a list if all items are removed
- On the course home page and search results, displays the image_src of each list's first item image src.

#### How should this be manually tested?
- Create a new empty list then search for it, it should not show up in results.
- Add 1 or more items to the list then search for it again, it should show up in results.
- The thumbnail for the list should be the same as the thumbnail for the 1st item in the list in search results.
- Favorite the list then go to the course home page and find it in the Favorites carousel, it should also have the same thumbnail.
- Remove all items from the list then search for it again, it should not show up in results.
